### DISCUSSION TITLE
Connecting CWL and ISA

### DISCUSSION THREAD

**Discussion by HLWeil** [nfdi4plants/ARC-specfication#121] [Comments and Suggestions]
- Include cwl workflows in software assays?
  -  `Protocol REF` column?
  
- How to reference and validate non-uploaded CWL Run results
  - CWL produces output, is used in next run or software assay, then deleted
    - This could also just be raw files downloaded via some tool (e.g. SRA)
  - Can we validate that it would exist, if cwl is run?

Note: Just collecting some thoughts from an in-person discussion (@Freymaurer @kMutagene @caroott)

**Comment by kMutagene** [comment]
For future reference, this initially came up while discussing the potential shortcomings of only allowing URLs that point to the exact resource for Input Nodes that point to external data. We imagined a potential solution for this where the actual accession of external data can be a workflow that downloads data which can then be referenced via local pattern, but then saw that this would violate the principle that local files MUST exist because in this solution, the workflow MUST run to produce the files.

However, this shows that this is a general problem we encounter, as this violation will always happen when ANY workflow output file is used in the ISA metadata - e.g. as input for an assay. A potential solution would be exceptions for workflow outputs for that rule.
